### PR TITLE
[tests-only] overwrite content of shared file

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds gui test for overwrite content of a file shared by sharer

### Related Issue
https://github.com/owncloud/client/issues/8651

### CODE REVIEW

File: test/gui/shared/steps/steps.py [hunk 1]
```diff
@@ -724,3 +735,19 @@ def step(context, resource, group):
 
     sharingDialog = SharingDialog()
     sharingDialog.selectCollaborator(group, True)
+
+
+@When('the user overwrites the file "|any|" with content "|any|"')
+def step(context, resource, content):
+    print("before overwrite")
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:742]: remove the print

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -724,3 +735,19 @@ def step(context, resource, group):
...
+    waitForFileToBeSynced(context, resource)
+    waitForFolderToBeSynced(context, '/')
+
+    snooze(3)
+
+    f = open(context.userData['clientSyncPath'] + resource, "w")
+    f.write(content)
+    f.close()
+
+    print("after overwrite")
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:752]: remove the print

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -724,3 +735,19 @@ def step(context, resource, group):
 
     sharingDialog = SharingDialog()
     sharingDialog.selectCollaborator(group, True)
+
+
+@When('the user overwrites the file "|any|" with content "|any|"')
+def step(context, resource, content):
+    print("before overwrite")
+    waitForFileToBeSynced(context, resource)
+    waitForFolderToBeSynced(context, '/')
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:744]: why wait for folder `/` to be synced? 🤔

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -724,3 +735,19 @@ def step(context, resource, group):
...
+
+    snooze(3)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:746]: can you write a comment above the `snooze(3)` that explains why this static wait is necessary?

File: test/gui/shared/steps/steps.py (hunk 3 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:744]: this is to sync everything in root

File: test/gui/shared/steps/steps.py (hunk 4 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:746]: That's a good idea. I'll add a comment and push it.

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:742]: it is for the log purpose that would be visible in CI as well

File: test/gui/shared/steps/steps.py (hunk 4 above)
Comment (individual-it) [reply test/gui/shared/steps/steps.py:750]: the problem is described in https://github.com/owncloud/client/issues/8832

It is an artificial delay set in the client. See also https://github.com/owncloud/client/blob/9117275c67ca311152b4f16749c6fd32c3d40fe9/src/libsync/syncengine.cpp#L61-L74

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (Talank) [reply test/gui/shared/steps/steps.py:742]: ok, then can you write a different message something like `starting file overwrite` and `file has been successfully overwritten`?

File: test/gui/shared/steps/steps.py (hunk 4 above)
Comment (individual-it) [reply test/gui/shared/steps/steps.py:750]: Please mention #8832 in the comment, that gives more background

File: test/gui/shared/steps/steps.py [hunk 5]
```diff
@@ -724,3 +735,23 @@ def step(context, resource, group):
...
+    # The client does not see the change although the changes have already been made thus we are having a race condition
+    # So for now we add 3sec static wait
+
+    snooze(3)
+
+    f = open(context.userData['clientSyncPath'] + resource, "w")
+    f.write(content)
+    f.close()
+
+    print("file has been overwritten")
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:756]: Do we need that debug line?

File: test/gui/shared/steps/steps.py [hunk 6]
```diff
@@ -724,3 +735,23 @@ def step(context, resource, group):
 
     sharingDialog = SharingDialog()
     sharingDialog.selectCollaborator(group, True)
+
+
+@When('the user overwrites the file "|any|" with content "|any|"')
+def step(context, resource, content):
+    print("starting file overwrite")
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:742]: Is this debug line needed?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/shared/steps/steps.py (hunk 5 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:756]: It's for log purpose in CI

**Review by individual-it — APPROVED** [review]

**Pull request merged**